components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoPermissionRequest.kt

```diff
@@ -64,6 +69,28 @@ sealed class GeckoPermissionRequest constructor(
                 PERMISSION_MEDIA_KEY_SYSTEM_ACCESS to Permission.ContentMediaKeySystemAccess()
             )
         }
+
+        @VisibleForTesting
+        internal var isCompleted = false
+            set(value) {
+                synchronized(this) {
```

Comment:
Hm don't you want the read and writes synchronized here? I am not sure why though? `grant` and `reject` should be called on the main thread only, right? What else is calling `isComplete`?

Reply:
You are right, my bad. I will remove it. We are only `granting` or `rejecting` on the main thread, and either way if we complete the gecko result in another thread it will be likely to throw an exception because of the looper issue.